lib/ProviderAws.js

```diff
@@ -129,25 +132,27 @@ module.exports = function(S) {
      */
 
     addProfileCredentialsImpl(credentials, prefix) { // separate profile environment variable prefix from obtaining credentials from the profile.
-      let profile = process.env[prefix + '_PROFILE'],
-        profileCredentials;
+      let profile = process.env[prefix + '_PROFILE'];
       if (profile) {
-        profileCredentials = this.getProfile(profile, true);
-        if (profileCredentials) {
-          if (profileCredentials.aws_access_key_id) {
-            credentials.accessKeyId = profileCredentials.aws_access_key_id;
-          }
-          if (profileCredentials.aws_secret_access_key) {
-            credentials.secretAccessKey = profileCredentials.aws_secret_access_key;
-          }
-          if (profileCredentials.aws_session_token) { // node.js aws-sdk standard
-            credentials.sessionToken = profileCredentials.aws_session_token;
-          }
-          if (profileCredentials.aws_security_token) { // python boto standard
-            credentials.sessionToken = profileCredentials.aws_security_token;
-          }
-        }
+        return this.getProfile(profile, true)
+          .then((profileCredentials) => {
+            if (profileCredentials) {
+              if (profileCredentials.aws_access_key_id) {
+                credentials.accessKeyId = profileCredentials.aws_access_key_id;
+              }
+              if (profileCredentials.aws_secret_access_key) {
+                credentials.secretAccessKey = profileCredentials.aws_secret_access_key;
+              }
+              if (profileCredentials.aws_session_token) { // node.js aws-sdk standard
+                credentials.sessionToken = profileCredentials.aws_session_token;
+              }
+              if (profileCredentials.aws_security_token) { // python boto standard
+                credentials.sessionToken = profileCredentials.aws_security_token;
+              }
+            }
+          });
       }
+      return BbPromise.resolve(null);
```

Comment:
An insignificant style suggestion but I'd suggest putting this return in an else clause.